src/main.rs

```diff
@@ -74,13 +80,14 @@ fn main() {
         }
 
         // Only update the gain if we know what the last reference was
-        match known {
-            true => known = !update_gain(current_gain, new_gain),
-            false => {
-                known = true;
-                current_gain = new_gain;
-            }
-        };
+        if known {
+            // Invalidate the known gain if we updated it, since it's not certain what miniDSP returns
```

Comment:
Reading the gain after an update is based on the fact that it's not certain that the miniDSP returns the exact gain that was written... it's kind of a black box, literally.

Reply:
Lawl. RIP